pennylane_lightning/core/_serialize.py

```diff
@@ -95,24 +109,39 @@ def rtype(self):
         """Real type."""
         return np.float32 if self.use_csingle else np.float64
 
+    @property
+    def sv_type(self):
+        """State vector matching ``use_csingle`` precision (and MPI if it is supported)."""
+        if self.use_mpi:
+            return self.statevectormpi_c128
+        return self.statevector_c128
+
     @property
     def named_obs(self):
         """Named observable matching ``use_csingle`` precision."""
+        if self.use_mpi:
+            return self.named_obsmpi_c64 if self.use_csingle else self.named_obsmpi_c128
         return self.named_obs_c64 if self.use_csingle else self.named_obs_c128
 
     @property
     def hermitian_obs(self):
         """Hermitian observable matching ``use_csingle`` precision."""
+        if self.use_mpi:
+            return self.hermitian_obsmpi_c64 if self.use_csingle else self.hermitian_obsmpi_c128
         return self.hermitian_obs_c64 if self.use_csingle else self.hermitian_obs_c128
 
     @property
     def tensor_obs(self):
         """Tensor product observable matching ``use_csingle`` precision."""
+        if self.use_mpi:
+            return self.tensor_prod_obsmpi_c64 if self.use_csingle else self.tensor_prod_obsmpi_c128
         return self.tensor_prod_obs_c64 if self.use_csingle else self.tensor_prod_obs_c128
 
     @property
     def hamiltonian_obs(self):
         """Hamiltonian observable matching ``use_csingle`` precision."""
+        if self.use_mpi:
+            return self.hamiltonianmpi_c64 if self.use_csingle else self.hamiltonianmpi_c128
         return self.hamiltonian_c64 if self.use_csingle else self.hamiltonian_c128
```

Comment:
I think we may want to revisit this class later, as I'd like to avoid a situation where we add if MPI everywhere, especially if we expect some of the backends to necessary use it. For now, let's keep as is.